benchmark.js

```diff
@@ -55,89 +61,227 @@ function blueTime(time) {
+              if (err) {
+                throw new Error('There was an error reading ' + info.gzFilePath);
+              }
+              info.gzSize = stats.size;
+              done();
+            });
+          });
+        },
+        // Open and read the size of the minified output
+        function (done) {
+          fs.stat(info.filePath, function (err, stats) {
+            if (err) {
+              throw new Error('There was an error reading ' + info.filePath);
+            }
+            info.size = stats.size;
+            done();
+          });
+        }
+      ], done);
+    }
 
-  // Open and read the size of the original input
-  fs.stat(filePath, function (err, stats) {
-    if (err) {
-      throw new Error('There was an error reading ' + filePath);
+    function testHTMLMinifier(done) {
+      // Begin timing after gzipped fixtures have been created
+      var startTime = Date.now();
+      var info = infos.minifier;
+      var args = [filePath, '-c', 'benchmark.conf', '-o', info.filePath];
+      fork('./cli', args).on('exit', function () {
+        minifiedTime = Date.now() - startTime;
+        readSizes(info, function () {
+          gzMinifiedTime = Date.now() - startTime;
+          done();
+        });
+      });
     }
-    var originalSize = stats.size;
 
-    exec('gzip --keep --force --best --stdout ' + filePath + ' > ' + gzFilePath, function () {
-      // Open and read the size of the gzipped original
-      fs.stat(gzFilePath, function (err, stats) {
+    function testMinimize(done) {
+      fs.readFile(filePath, function (err, data) {
         if (err) {
-          throw new Error('There was an error reading ' + gzFilePath);
+          throw new Error('There was an error reading ' + filePath);
         }
-
-        var gzOriginalSize = stats.size;
-
-        // Begin timing after gzipped fixtures have been created
-        var startTime = new Date();
-        exec('node ' + command, function () {
-
-          // Open and read the size of the minified output
-          fs.stat(minifiedFilePath, function (err, stats) {
+        minimize.parse(data, function (error, data) {
+          var info = infos.minimize;
+          fs.writeFile(info.filePath, data, function (err) {
             if (err) {
-              throw new Error('There was an error reading ' + minifiedFilePath);
+              throw new Error('There was an error writing ' + info.filePath);
             }
+            readSizes(info, done);
+          });
+        });
+      });
+    }
 
-            var minifiedSize = stats.size;
-            var minifiedTime = new Date() - startTime;
-
-            minimize.parse(
-              fs.readFileSync(filePath),
-              function (error, data) {
-                console.log('minimize',
-                  filePath,
-                  toKb(data.length),
-                  toKb(minifiedSize));
+    function testWillPeavy(done) {
+      fs.readFile(filePath, {
```

Comment:
Definitely looks much better, thanks! One final nit though is that we can probably still cut down on repetition a bit more here. There's all this similar boilerplate with reading file, throwing error, setting headers and so on, which we can extract into a generic method. It will help with accidentally changing something in one location (e.g. testWillPeavy) and forgetting to change another. And it will make it much easier to test other pages if the need arises. Or is there too much differences between them?